src/main/webapp/app/entities/flair-bi/flair-bi.controller.js

```diff
@@ -604,7 +606,7 @@
             );
             v.isCardRevealed = v.isCardRevealed == undefined ? true : !v.isCardRevealed;
             if (!v.isCardRevealed) {
-                $rootScope.$broadcast("flairbiApp:onData-open");
+                $rootScope.$broadcast("flairbiApp:onData-open",v);
```

Comment:
no need to send v in broadcast. it will impact the performance...use VisualDispatchService.setVisual and VisualDispatchService.getVisual to set/get the visualisation

Reply:
done